src/xcode/ENA/ENA/Source/Services/Risk/Provider/Revocation/__tests__/RevocationProviderTests.swift

```diff
@@ -0,0 +1,239 @@
+					}
+				),
+				// Update KID-Type chunk for recovery certificate, recovery certificate is revoked
+				LoadResource(
+					result: .success(try revocationChunk()),
+					willLoadResource: { resource in
+						guard let resource = resource as? KIDTypeChunkResource else {
+							XCTFail("wrong resource type, KIDTypeChunkResource expected")
+							return
+						}
+
+						XCTAssertEqual(resource.locator.paths, ["version", "v1", "dcc-rl", "\(self.recoveryCertificateKeyIdentifier)0a", "e3", "d6", "chunk"])
+					}
+				),
+				// As recovery certificate is revoked after checking with hash type 0a already, 0b KID-Type index update is skipped for recovery certificate
+				// Continue with update of KID-Type index for booster certificate key identifier, hash type 0b
+				LoadResource(
+					result: .success(try kidTypeIndex()),
+					willLoadResource: { resource in
+						guard let resource = resource as? KIDTypeIndexResource else {
+							XCTFail("wrong resource type")
+							return
+						}
+
+						XCTAssertEqual(resource.locator.paths, ["version", "v1", "dcc-rl", "\(self.boosterCertificateKeyIdentifier)0b", "index"])
+					}
+				),
+				// Update KID-Type chunk for booster certificate, booster certificate is not revoked for hash type 0b
+				LoadResource(
+					result: .success(try revocationChunk()),
+					willLoadResource: { resource in
+						guard let resource = resource as? KIDTypeChunkResource else {
+							XCTFail("wrong resource type, KIDTypeChunkResource expected")
+							return
+						}
+
+						XCTAssertEqual(resource.locator.paths, ["version", "v1", "dcc-rl", "\(self.boosterCertificateKeyIdentifier)0b", "70", "12", "chunk"])
+					}
+				),
+				// Update of KID-Type index for booster certificate key identifier, hash type 0c
+				LoadResource(
+					result: .success(try kidTypeIndex()),
+					willLoadResource: { resource in
+						guard let resource = resource as? KIDTypeIndexResource else {
+							XCTFail("wrong resource type")
+							return
+						}
+
+						XCTAssertEqual(resource.locator.paths, ["version", "v1", "dcc-rl", "\(self.boosterCertificateKeyIdentifier)0c", "index"])
+					}
+				),
+				// Update KID-Type chunk for booster certificate, booster certificate is revoked for hash type 0c
+				LoadResource(
+					result: .success(try revocationChunk()),
+					willLoadResource: { resource in
+						guard let resource = resource as? KIDTypeChunkResource else {
+							XCTFail("wrong resource type, KIDTypeChunkResource expected")
+							return
+						}
+
+						XCTAssertEqual(resource.locator.paths, ["version", "v1", "dcc-rl", "\(self.boosterCertificateKeyIdentifier)0c", "25", "89", "chunk"])
+					}
+				)
+			]
+		)
+
+		let revocationProvider = RevocationProvider(restService, signatureVerifier: MockVerifier())
+
+		let expectation = expectation(description: "success expectation")
+
+		revocationProvider.updateCache(with: certificates) {
```

Comment:
Is it not needed to check if the correct data is in the cache after updating it?

Reply:
The goal of this test from our point of view was to check that the correct requests are done in the correct order. Assuming the RestServiceProvider works correctly (which I guess is tested somewhere else) we then know that the cache will be put-to-date. (And as the `RestServiceProviderStub` has no working cache implementation we cannot easily test this and would only test the stub implementation anyway.)